lib/puppet/defaults.rb

```diff
@@ -34,6 +34,11 @@ def self.default_diffargs
   )
 
   define_settings(:main,
+    :puppetdir => {
+        :default  => nil,
+        :type     => :directory,
+        :desc     => "TODO ncw"
```

Comment:
@nfagerlund Not sure how we want to describe these directories now.  Suggestions welcome :smiley: